hawaii-logging/src/main/java/org/hawaiiframework/logging/http/DefaultHawaiiRequestResponseLogger.java

```diff
@@ -0,0 +1,205 @@
+
+    /**
+     * The configuration to use.
+     */
+    private final HawaiiLoggingConfigurationProperties configuration;
+
+    public DefaultHawaiiRequestResponseLogger(final HttpRequestResponseLogUtil httpRequestResponseLogUtil,
+            final HawaiiLoggingConfigurationProperties configuration) {
+        this.httpRequestResponseLogUtil = httpRequestResponseLogUtil;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public void logRequest(final HttpRequest request, final byte[] body) {
+        try (AutoCloseableKibanaLogField callMethod = KibanaLogFields.tagCloseable(CALL_METHOD, request.getMethodValue());
+                AutoCloseableKibanaLogField kibanaLogField = KibanaLogFields.logType(CALL_REQUEST_BODY)) {
+
+            final String contentType = getContentType(request);
+            if (contentTypeCanBeLogged(contentType)) {
+                logContent(request, body);
+            }
+        }
+    }
+
+    @Override
+    public void logRequest(final ResettableHttpServletRequest wrappedRequest) throws IOException {
+        final String requestUri = wrappedRequest.getRequestURI();
+        final int contentLength = wrappedRequest.getContentLength();
+        final String contentType = wrappedRequest.getContentType();
+
+        try (AutoCloseableKibanaLogField kibanaLogField = KibanaLogFields.logType(REQUEST_BODY)) {
+            LOGGER.info("Invoked '{}' with content type '{}' and size of '{}' bytes.", requestUri, contentType, contentLength);
+            try {
+                if (contentTypeCanBeLogged(contentType)) {
+                    logContent(wrappedRequest);
+                }
+            } finally {
+                wrappedRequest.reset();
+            }
+        }
+    }
+
+    @Override
+    public void logResponse(final ClientHttpResponse response) throws IOException {
+        if (contentTypeCanBeLogged(getContentType(response))) {
+            logContent(response);
+        }
+    }
+
+    @Override
+    public void logResponse(final HttpServletRequest servletRequest,
+            final ContentCachingWrappedResponse wrappedResponse)
+            throws IOException {
+
+        final String request = httpRequestResponseLogUtil.getRequestUri(servletRequest);
+        final int contentLength = wrappedResponse.getContentSize();
+        final String contentType = wrappedResponse.getContentType();
+        final HttpStatus httpStatus = HttpStatus.valueOf(wrappedResponse.getStatus());
+
+        KibanaLogFields.set(HTTP_STATUS, httpStatus.value());
+        try (AutoCloseableKibanaLogField kibanaLogField = KibanaLogFields.logType(RESPONSE_BODY)) {
+            LOGGER.info("Response '{}' is '{} {}' with content type '{}' and size of '{}' bytes.", request,
+                    httpStatus.value(), httpStatus.getReasonPhrase(), contentType, contentLength);
+            if (contentTypeCanBeLogged(contentType)) {
+                logContent(servletRequest, wrappedResponse);
```

Comment:
Why switch from log response to log content?

Reply:
At first I wanted to create one single logContent method that logged all the responses/request but that wasn't a nice solution. Removed the logContent method an just use logger.info instead.